nikola/plugin_manager.py

```diff
@@ -0,0 +1,263 @@
+                self.has_warnings = True
+                continue
+
+            plugin_class = plugin_classes[0]
+
+            if not issubclass(plugin_class, CATEGORIES[candidate.category]):
+                self.logger.warning(f"{plugin_id} has category '{candidate.category}' in the .plugin file, but the implementation class {plugin_class} does not inherit from this category - plugin will not be loaded")
+                self.has_warnings = True
+                continue
+
+            try:
+                plugin_object = plugin_class()
+            except Exception:
+                self.logger.exception(f"{plugin_id} threw an exception while creating the instance")
+                self.has_warnings = True
+                continue
+            self.logger.debug(f"Loaded {plugin_id}")
+            info = PluginInfo(
+                name=name,
+                description=candidate.description,
+                plugin_id=candidate.plugin_id,
+                category=candidate.category,
+                compiler=candidate.compiler,
+                source_dir=source_dir,
+                module_name=module_name,
+                module_object=module_object,
+                plugin_object=plugin_object,
+            )
+            self.plugins.append(info)
+
+        self._plugins_by_category = {category: [] for category in CATEGORY_NAMES}
+        for plugin_info in self.plugins:
+            self._plugins_by_category[plugin_info.category].append(plugin_info)
+
+        if self.has_warnings:
+            self.logger.warning("Some plugins failed to load. Please review the above warning messages.")
+            # TODO remove following messages and delay in v8.3.1
+            self.logger.warning("You may need to update some plugins (from plugins.getnikola.com) or to fix their .plugin files.")
+            self.logger.warning("Waiting 2 seconds before continuing.")
+            time.sleep(2)
+
+    def get_plugins_of_category(self, category: str) -> List[PluginInfo]:
+        """Get loaded plugins of a given category."""
+        return self._plugins_by_category.get(category, [])
+
+    def get_plugin_by_name(self, name: str, category: Optional[str] = None) -> Optional[PluginInfo]:
+        """Get a loaded plugin by name and optionally by category. Returns None if no such plugin is loaded."""
+        for p in self.plugins:
+            if p.name == name and (category is None or p.category == category):
+                return p
+
+    # Aliases for Yapsy compatibility
+    # TODO: remove in v9
+    def getPluginsOfCategory(self, category: str) -> List[PluginInfo]:
+        """Get loaded plugins of a given category."""
+        self.logger.warning("Legacy getPluginsOfCategory method was used, it may be removed in the future. Please change it to get_plugins_of_category.")
+        return self._plugins_by_category.get(category, [])
+
+    def getPluginByName(self, name: str, category: Optional[str] = None) -> Optional[PluginInfo]:
```

Comment:
Should this also get a `TODO: remove in v9` comment?

Reply:
I intended for the first one to apply to both, but having two comments doesn’t hurt and would be clearer, so I added it here too.